bicytok/test/test_model.py

```diff
@@ -27,3 +27,80 @@ def test_optimize_design():
         valencies=np.array([[2, 2]]),
         prevOptAffs=[8.0, 8.0],
     )
+"""
+
+
+def sample_data():
+    np.random.seed(0)
+    recAbundances = np.random.rand(100, 10) * 10
+    targ = np.random.choice([True, False], size=100, p=[0.3, 0.7])
+    offTarg = ~targ
+    return recAbundances, targ, offTarg
+
+
+def test_KL_EMD_1D():
+    recAbundances, targ, offTarg = sample_data()
+    targ = np.array(targ, dtype=bool)
+    offTarg = np.array(offTarg, dtype=bool)
+
+    KL_div_vals, EMD_vals = KL_EMD_1D(recAbundances, targ, offTarg)
+
+    assert len(KL_div_vals) == recAbundances.shape[1]
+    assert len(EMD_vals) == recAbundances.shape[1]
+    print([type(i) for i in np.append(targ, offTarg)])
+
+
+def test_KL_EMD_2D():
+    recAbundances, targ, offTarg = sample_data()
+    targ = np.array(targ, dtype=bool)
+    offTarg = np.array(offTarg, dtype=bool)
+    KL_div_vals, EMD_vals = KL_EMD_2D(recAbundances, targ, offTarg)
+
+    assert KL_div_vals.shape == (recAbundances.shape[1], recAbundances.shape[1])
+    assert EMD_vals.shape == (recAbundances.shape[1], recAbundances.shape[1])
+    assert np.all(np.isnan(KL_div_vals) | (KL_div_vals >= 0))
+    assert np.all(np.isnan(EMD_vals) | (EMD_vals >= 0))
+
+
+def test_invalid_inputs():
+    recAbundances = np.random.rand(100, 10)
+    targ = np.random.choice([True, False], size=100, p=[0.3, 0.7])
+    offTarg = ~targ
+
+    # Test invalid inputs for KL_EMD_1D
+    try:
```

Comment:
https://stackoverflow.com/questions/23337471/how-do-i-properly-assert-that-an-exception-gets-raised-in-pytest